pkg/pod/list.go

```diff
@@ -105,3 +144,48 @@ func WaitForAllPodsInNamespaceRunning(
 
 	return true, nil
 }
+
+// WaitForThePodReplicasCountInNamespace waiting for the specific pod replicas count in the namespace that match options.
+func WaitForThePodReplicasCountInNamespace(
```

Comment:
Can you please share use case for this function. Do you check if all pods under specific deployment is equal to specific replica count?

Reply:
I have many, mostly related to the app verifications. Anyway, one of the service mesh examples:

```go


				glog.V(100).Info("Check the service-mesh control plane pods are creating on the " +
					"istio-system project")
				glog.V(100).Infof("Confirm that %s pod was deployed and running in %s namespace",
					istioBasicPodLabel, istioNamespace)
				_, err = pod.WaitForThePodReplicasCountInNamespace(
					APIClient,
					istioNamespace,
					v1.ListOptions{LabelSelector: istioBasicPodLabel},
					3,
					5*time.Second)
```

Reply:
the filter-out by name pattern and len() is not suitable in this case for example, cause first I need to check we have specific number of pods and then we start to wait until all pods Running

Reply:
The idea is that if you application is **daemonset** then you can pull it using **Pull** method from **daemonset** package and then re-use **Ready** method https://github.com/openshift-kni/eco-goinfra/blob/5692432da6a7010b50d866cc64a46b99596ccb6a/pkg/daemonset/daemonset.go#L334C6-L334C6 or daemonset 

If you application is **deployment** then you can deployment's Pull  method and **Ready** method https://github.com/openshift-kni/eco-goinfra/blob/5692432da6a7010b50d866cc64a46b99596ccb6a/pkg/deployment/deployment.go#L423C5-L423C5

Will this part of ready code work for you ?

Reply:
@kononovn maybe in this particular case it will help. But if we are going to transfer our regression from python to go, I know we have more applications (not daemonset) related cases. I'm not sure what is a problem to have method which return number of ready replicas?
If you don't want to have it on infra, I could move it to the my private internal on eco-gotest

Reply:
Hey Elena, I think Nikita's point is that it is a bit redundant with code that is already written. Replicas are defined at deployment, replicaset, statefulset, daemonset level. It's better to just use those pkgs for tracking if your pods are ready or not. Alternatively you can just use `WaitForAllPodsInNamespaceRunning` if you have multiple pods in a namespace with different owners.

Reply:
If you still feel that these options don't fit your use case, feel free to add it to your team's internal pkgs

Reply:
1. I can not use WaitForAllPodsInNamespaceRunning, this is a point. I wait for the particular number of pods to run and WaitForAllPodsInNamespaceRunning returns true when all pods are running, but ignoring number of pods.
2. statefulset is not suitable in case of app
3. deployment -> IsReady: 
       - maybe it could be used, anyway, IMHO you need to change the name to the Wait... If method has a timeout and wait for specific amount of time - it should be clear from the method name 
       - and it wouldn't help in the following case:
       
```go
[kni@registry.kni-qe-3 ~]$ oc get all -n benchmark-operator
Warning: apps.openshift.io/v1 DeploymentConfig is deprecated in v4.14+, unavailable in v4.10000+
NAME                                                READY   STATUS    RESTARTS   AGE
pod/backpack-fecfce25-2rtlm                         1/1     Running   0          6m4s
pod/backpack-fecfce25-4wwst                         1/1     Running   0          6m4s
pod/backpack-fecfce25-6cr55                         1/1     Running   0          6m4s
pod/backpack-fecfce25-7ftrj                         1/1     Running   0          6m4s
pod/backpack-fecfce25-9tnxp                         1/1     Running   0          6m4s
pod/backpack-fecfce25-sjfvv                         1/1     Running   0          6m4s
pod/backpack-fecfce25-t8mnz                         1/1     Running   0          6m4s
pod/benchmark-controller-manager-59cfd7b7b7-7mh2j   2/2     Running   0          6m45s
pod/uperf-client-10.130.2.22-fecfce25-hrjph         1/1     Running   0          4m40s
pod/uperf-client-10.130.2.23-fecfce25-85qzr         1/1     Running   0          4m35s
pod/uperf-client-10.130.2.24-fecfce25-bc986         1/1     Running   0          4m37s
pod/uperf-client-10.130.2.25-fecfce25-gp9fs         1/1     Running   0          4m44s
pod/uperf-client-10.130.2.26-fecfce25-vj8kl         1/1     Running   0          4m43s
pod/uperf-client-10.130.2.27-fecfce25-s9mjr         1/1     Running   0          4m42s
pod/uperf-client-10.130.2.29-fecfce25-dcjvc         1/1     Running   0          4m33s
pod/uperf-client-10.130.2.30-fecfce25-n5rxn         1/1     Running   0          4m39s
pod/uperf-client-10.130.2.31-fecfce25-t9tdt         1/1     Running   0          4m42s
pod/uperf-client-10.130.2.32-fecfce25-dwkp6         1/1     Running   0          4m34s
pod/uperf-client-10.130.2.33-fecfce25-kt5n2         1/1     Running   0          4m39s
pod/uperf-client-10.130.2.34-fecfce25-q78lf         1/1     Running   0          4m32s
pod/uperf-client-10.130.2.35-fecfce25-94t9h         1/1     Running   0          4m36s
pod/uperf-client-10.130.2.36-fecfce25-5lpgh         1/1     Running   0          4m38s
pod/uperf-client-10.130.2.37-fecfce25-xfnpb         1/1     Running   0          4m41s
pod/uperf-client-10.130.2.38-fecfce25-6qwcq         1/1     Running   0          4m36s
pod/uperf-client-10.130.2.39-fecfce25-jw65n         1/1     Running   0          4m45s
pod/uperf-client-10.130.2.40-fecfce25-f4v2s         1/1     Running   0          4m44s
pod/uperf-client-10.130.2.41-fecfce25-s22wr         1/1     Running   0          4m33s
pod/uperf-client-10.130.2.42-fecfce25-m77rl         1/1     Running   0          4m31s
pod/uperf-server-0-fecfce25-g6p85                   1/1     Running   0          5m27s
pod/uperf-server-1-fecfce25-vp4jd                   1/1     Running   0          5m26s
pod/uperf-server-10-fecfce25-6bfz8                  1/1     Running   0          5m20s
pod/uperf-server-11-fecfce25-rv8lx                  1/1     Running   0          5m20s
pod/uperf-server-12-fecfce25-7dttq                  1/1     Running   0          5m19s
pod/uperf-server-13-fecfce25-wdwwl                  1/1     Running   0          5m18s
pod/uperf-server-14-fecfce25-rl4xx                  1/1     Running   0          5m18s
pod/uperf-server-15-fecfce25-qqql7                  1/1     Running   0          5m17s
pod/uperf-server-16-fecfce25-b4sl8                  1/1     Running   0          5m16s
pod/uperf-server-17-fecfce25-x96q2                  1/1     Running   0          5m16s
pod/uperf-server-18-fecfce25-vr4bm                  1/1     Running   0          5m15s
pod/uperf-server-19-fecfce25-kkdfz                  1/1     Running   0          5m15s
pod/uperf-server-2-fecfce25-8dljv                   1/1     Running   0          5m25s
pod/uperf-server-3-fecfce25-rqk9q                   1/1     Running   0          5m25s
pod/uperf-server-4-fecfce25-xr9lj                   1/1     Running   0          5m24s
pod/uperf-server-5-fecfce25-rgmsh                   1/1     Running   0          5m23s
pod/uperf-server-6-fecfce25-jprt2                   1/1     Running   0          5m23s
pod/uperf-server-7-fecfce25-jdxdh                   1/1     Running   0          5m22s
pod/uperf-server-8-fecfce25-sw7c6                   1/1     Running   0          5m22s
pod/uperf-server-9-fecfce25-9fh9z                   1/1     Running   0          5m21s

NAME                               DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/backpack-fecfce25   7         7         7       7            7           <none>          6m4s

NAME                                           READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/benchmark-controller-manager   1/1     1            1           6m45s

NAME                                                      DESIRED   CURRENT   READY   AGE
replicaset.apps/benchmark-controller-manager-59cfd7b7b7   1         1         1       6m45s

```

Reply:
@elenagerman Thanks for clarifying. I think we want to encourage folks using the project to use higher-level structs for tracking pod replica counts, so this scenario doesn't really fit into infra. Can you please move it to your team's internal pkgs? Thanks